homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -44,65 +63,77 @@ class SurePetcareBinarySensor(BinarySensorDevice):
     """A binary sensor implementation for Sure Petcare Entities."""
 
     def __init__(
-        self, _id: int, name: str, spc, device_class: str, sure_type: SureThingID
+        self: BinarySensorDevice,
+        _id: int,
+        spc: SurePetcareAPI,
+        device_class: str,
+        sure_type: SureProductID,
     ):
         """Initialize a Sure Petcare binary sensor."""
         self._id = _id
-        self._name = name
-        self._spc = spc
-        self._device_class = device_class
         self._sure_type = sure_type
-        self._state = {}
+        self._device_class = device_class
+
+        self._spc: SurePetcareAPI = spc
+        self._spc_data: Dict[str, Any] = self._spc.states[self._sure_type].get(self._id)
```

Comment:
Does it matter that we'll replace the spc data in the spc instance on every update? Ie, that `self._spc_data` here might become stale.